Source/udSocket.cpp

```diff
@@ -370,8 +370,7 @@ udResult udSocket_Open(udSocket **ppSocket, const char *pAddress, uint32_t port,
         UD_ERROR_SET(udR_InternalCryptoError);
       }
 
-      // TODO: Has to be removed before we ship...
-      mbedtls_ssl_conf_authmode(&pSocket->tlsClient.conf, MBEDTLS_SSL_VERIFY_NONE);
+      mbedtls_ssl_conf_authmode(&pSocket->tlsClient.conf, MBEDTLS_SSL_VERIFY_REQUIRED);
```

Comment:
Add a new flag for `udSCF_DisableVerification` and then:
```suggestion
      if ((flags & udSCF_DisableVerification) == 0)
        mbedtls_ssl_conf_authmode(&pSocket->tlsClient.conf, MBEDTLS_SSL_VERIFY_REQUIRED);
```
This will allow you to modify the unit tests to disable verification for the local tests but keep verification working for external tests.